crates/storage/lib.rs

```diff
@@ -18,3 +18,6 @@ pub use trie_db::layering::apply_prefix;
 /// Store Schema Version, must be updated on any breaking change
 /// An upgrade to a newer schema version invalidates currently stored data, requiring a re-sync.
 pub const STORE_SCHEMA_VERSION: u64 = 1;
+
+/// Name of the file storing the metadata about the database
+pub const STORE_METADATA_FILENAME: &str = "metadata.json";
```

Comment:
I think the place for the schema versioning is the DB. Decoupling it is asking for trouble. E.g., I don't see the `removedb` command deleting this properly.
If we don't want it to be data, we may follow a scheme similar to what Postgres does, which is encoding versioning in file names (e.g., the internal directory might be `data-{schema_version}`, with failure indicating incompatibility).